src/components/contact-us/contact-us-controller.ts

```diff
@@ -62,27 +65,58 @@ export function contactUsGet(req: Request, res: Response): void {
   if (req.query.supportType === SUPPORT_TYPE.GOV_SERVICE) {
     return res.render("contact-us/index-gov-service-contact-us.njk");
   }
-  let referer = req.headers.referer;
+
+  let referer = validateReferer(req.headers.referer);
 
   if (req.query.referer) {
-    referer = req.query.referer as string;
+    referer = validateReferer(req.query.referer as string);
+    logger.info(`referer with referer query param ${referer}`);
   }
 
   if (req.headers.referer && req.headers.referer.includes("referer")) {
-    const urlObj = new URL(req.headers.referer);
-    referer = urlObj.searchParams.get("referer");
+    try {
+      referer = validateReferer(
+        new URL(req.headers.referer).searchParams.get("referer")
+      );
+      logger.info(`referer with referer header param ${referer}`);
+    } catch {
+      logger.warn(
+        `unable to parse referer with referer param ${req.headers.referer}`
+      );
+    }
   }
 
   return res.render("contact-us/index-public-contact-us.njk", {
     referer: referer,
   });
 }
 
+function validateReferer(referer: string): string {
+  let valid = false;
+  let url;
+  try {
+    if (CONTACT_US_REFERER_ALLOWLIST.includes(referer)) {
+      valid = true;
+    } else {
+      url = new URL(xss(referer));
```

Comment:
What is `xss` doing for us here? If it's just stripping HTML tags, does the new URL constructor not also sanitise the input in an equivalent way (genuines question, but I assume HTML tags don't survive URL encoding anyway?)